### PR TITLE
 Add a sample application exporting Prometheus metrics.

### PR DESCRIPTION
**Pull request by jm-franc**
## Description

This PR adds a sample that includes a simple application that generates
Prometheus metrics and exports them to Google Cloud Monitoring using Google
Cloud Managed Service for Prometheus.

It supersedes the 'prometheus-to-sd' sample which uses the deprecated
'prometheus-to-sd' sidecar container. The simple application generating
Prometheus metrics (and its build configuration) has been moved to this
sample.

## Tasks

* [ ] The [contributing guide](https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/blob/main/.github/CONTRIBUTING.md) has been read and followed.
* [ ] The samples added / modified have been fully tested.
* [ ] Workflow files have been added / modified, if applicable.
* [ ] Region tags have been properly added, if new samples.
* [ ] All dependencies are set to up-to-date versions, as applicable.

### CODE REVIEW

File: .github/terraform/google-cloud-build-triggers.tf [hunk 1]
```diff
@@ -71,11 +87,11 @@ resource "google_cloudbuild_trigger" "custom-metrics-direct-to-sd" {
     }
 }
 
-resource "google_cloudbuild_trigger" "custom-metrics-prometheus-to-sd" {
-    name = "kubernetes-engine-samples-custom-metrics-prometheus-to-sd"
+resource "google_cloudbuild_trigger" "custom-metrics-prometheus-to-sd-gmp" {
+    name = "kubernetes-engine-samples-custom-metrics-prometheus-to-sd-gmp"
```
Comment (laoj2) [inline .github/terraform/google-cloud-build-triggers.tf:91]: a nit here and below (including the directories): `custom-metrics-gmp` 

We can remove the references to prometheus-to-sd altogether (including the naming) in favor of GMP.

File: observability/custom-metrics-autoscaling/prometheus-to-sd-gmp/custom-metrics-prometheus-sd-gmp.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+metadata:
+  name: "prometheus-to-prometheus"
+spec:
+  selector:
+    matchLabels:
+      run: "custom-metric-prometheus-sd-gmp"
+  endpoints:
+  - port: 8080
+    path: /metrics
+    interval: 60s
```
Comment (laoj2) [inline observability/custom-metrics-autoscaling/prometheus-to-sd-gmp/custom-metrics-prometheus-sd-gmp.yaml:58]: I know that this demo application exports a constant value, but what about setting this interval to a shorter period (e.g., 15s)? So people can have a faster autoscaling reaction time if they decide to customize the underlying application and reuse this PodMonitoring object.

Comment (jm-franc) [reply observability/custom-metrics-autoscaling/prometheus-to-sd-gmp/custom-metrics-prometheus-sd-gmp.yaml]: Done.

File: .github/terraform/google-cloud-build-triggers.tf (hunk 1 above)
Comment (jm-franc) [reply .github/terraform/google-cloud-build-triggers.tf]: Done. (As discussed oob, I named the directory `google-managed-prometheus`, and the file `custom-metrics-gmp.yaml`.)

File: observability/custom-metrics-autoscaling/google-managed-prometheus/custom-metrics-gmp.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        - --metric-value=40
+        - --port=8080
+# [END container_custom_metrics_prometheus_deployment]
+# [END gke_google_managed_prometheus_custom_metrics_gmp_deployment_custom_metrics_gmp]
+---
+# [START gke_google_managed_prometheus_custom_metrics_gmp_pod_monitoring_custom_metrics_gmp]
+apiVersion: monitoring.googleapis.com/v1
+kind: PodMonitoring
+metadata:
+  name: "prometheus-to-prometheus"
```
Comment (laoj2) [inline observability/custom-metrics-autoscaling/google-managed-prometheus/custom-metrics-gmp.yaml:50]: what about calling this `custom-metrics-exporter` ?

**Review by laoj2 — APPROVED** [review]

File: observability/custom-metrics-autoscaling/google-managed-prometheus/custom-metrics-gmp.yaml (hunk 3 above)
Comment (jm-franc) [reply observability/custom-metrics-autoscaling/google-managed-prometheus/custom-metrics-gmp.yaml:50]: Done.

**Review by bourgeoisor — APPROVED** [review]

File: observability/custom-metrics-autoscaling/google-managed-prometheus/custom-metrics-gmp.yaml (hunk 3 above)
Comment (jm-franc) [reply observability/custom-metrics-autoscaling/google-managed-prometheus/custom-metrics-gmp.yaml:50]: Fixed in https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/pull/1334

**Pull request merged**